### ISSUE TITLE
Version 3.0 progress

### ISSUE THREAD

**Issue by TimSirmovics** [roadkillwiki/roadkill#79]
Hi, just wondering what the progress is for v3.
Is it under development?

**Comment by yetanotherchris** [comment]
Yes it is! It was frozen while .NET Core 2 came out, which annoyingly would've made a lot of the conversion problems a lot easier. Currently there's a few issues with converting Lucene and waiting for a stable version of that to exist with .NET Core. 

I'm currently working on two other projects however, Syringe .NET core conversion and another private one. I'm aiming to use the learnings from Syringe however to speed up the Roadkill conversion.

Because of the size of the codebase, the conversion has been set back quite a lot by Microsoft's indecisiveness over .NET core, but I think it'll now be on track with .NET Core 2.

**Comment by satano** [comment]
Hello

@yetanotherchris you have added **help wanted** tag. It would be nice to write how can people help - what exactly to do.

**Comment by yetanotherchris** [comment]
Wrong label, my mistake!

**Comment by zunath** [comment]
Do you happen to have any update regarding this?

**Comment by yetanotherchris** [comment]
The current status is:

- I'm working on a project I can't talk about, which is consuming all my spare time.
- Part of this is [nooptime](github.com/yetanotherchris/nooptime/)
- I am working on a lot of Docker-based projects right now

All of this work is going to make the Roadkill conversion to .NET Core a lot faster. The majority of the Roadkill work has been moving away from Creole, which is now pretty much done. Creole was understandably a popular choice, but as a markdown language sucks.

Time frames: I'm not sure, but soon (TM) 😄 

I'm really keen to get started on Roadkill 3 and get people developing for it, and watch it become incredibly fast and cheap to deploy with Docker. But as mentioned, I have commercial projects that take precedence right now.

**Comment by zunath** [comment]
Thanks for the response. Totally understand! Thanks for your continued support.

**Comment by kashcode** [comment]
bump

**Comment by kendallb** [comment]
Any updates on this? I am looking around for a .net based wiki engine we can use for our documentation, and stumbled upon this. Would prefer to spend time with v3 if it is being actively developed. Is there a branch of the code anywhere?

**Comment by yetanotherchris** [comment]
Yes, on the dotnetcore branch. It's currently just RESTFul API, largely done (users and search left) with the UI being transitioned to use it next. 

I'll probably do this server side to begin with and then move to make it a React or VueJS UI later.

Hope that helps

**Comment by kendallb** [comment]
Ok thanks. If we start with v2, is there an upgrade path to v3?

**Comment by yetanotherchris** [comment]
Yes, there will be a tool to migrate the database. The configuration will be manual but should be straight forward
